app/src/main/java/com/example/bookwormadventuresdeluxe2/AddOrEditBooksActivity.java

```diff
@@ -105,6 +107,36 @@ public void saveBook(View view)
         }
     }
 
+    /**
+     * Validate the fields entered in this activity
+     * Title, author, and isbn cannot be empty
+     *
+     * @return true if all fields are valid, false otherwise
+     */
+    private boolean fieldsValid()
+    {
+        boolean valid = true;
+        if (TextUtils.isEmpty(titleView.getText().toString()))
+        {
+            EditTextValidator.isEmpty(titleView);
+            valid = false;
+        }
+        if (TextUtils.isEmpty(authorView.getText().toString()))
+        {
+            EditTextValidator.isEmpty(authorView);
+            valid = false;
+        }
+        /*
+        // Leaving this in in case the requirements include the ISBN to be non-empty
```

Comment:
I'm unsure on the requirement for this, but maybe we could make it for now either be allowed to be empty or the length of however many digits the isbn should be? So you could either leave it empty or n digits? Then later we can remove the empty part if we find out that isbn should always be included